nemo_skills/evaluation/top_answer_processor.py

```diff
@@ -0,0 +1,314 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import shutil
+import sys
+from collections import Counter, defaultdict
+from itertools import zip_longest
+from pathlib import Path
+from typing import Any, List, Tuple
+from enum import Enum
+
+import hydra
+from omegaconf import MISSING
+from tqdm import tqdm
+
+from nemo_skills.code_execution.math_grader import extract_answer
+from nemo_skills.evaluation.metrics import read_predictions
+from nemo_skills.utils import get_help_message, nested_dataclass, setup_logging, unroll_files
+
+LOG = logging.getLogger(__file__)
+
+
+@nested_dataclass(kw_only=True)
+class ProcessTopAnswerConfig:
+    """Top-level parameters for the script"""
+
+    # Input_dir relative to which all the input_files are specified
+    input_dir: str = MISSING
```

Comment:
not related to your changes, but it's best to not use MISSING, since it's only properly checked by hydra and all of our code can be used from python directly now. Instead just have those args on top without any default value